MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -2789,7 +2789,7 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 #if (MVK_IOS || MVK_MACOS) && MVK_XCODE_12
 	if (supportsMTLGPUFamily(Apple7)) { gpuFam = MTLGPUFamilyApple7; }
 #endif
-#if MVK_IOS && MVK_XCODE_13
+#if (MVK_IOS || MVK_MACOS) && MVK_XCODE_13
```

Comment:
Are we confident that the Xcode 13 macOS SDK supports `MTLGPUFamilyApple8` (in addition to the current Xcode 14 SDK)? I'm concerned that a build using Xcode 13 might fail, otherwise.

And if so, does Xcode 13 tvOS SDK also support `MTLGPUFamilyApple8`? In that case, we could change this to just `#if MVK_XCODE_13`.

If we don't know these, perhaps we could make this `#if MVK_XCODE_14 || (MVK_IOS && MVK_XCODE_13)`?

Reply:
According to the docs `MTLGPUFamilyApple8` was available with iOS 13 and macOS 10.15. The Xcode 13.0 release apparently supported iOS 15 and macOS 11.3. Though I've seen the Xcode 11.7 CI (which should support at least iOS 13 and macOS 10.15) fail finding even Apple6 and Apple7. The headers for MTLGPUFamily don't even have a `API_AVAILABLE` macro on them, so I can't properly determine it like that. However, I checked this [github repo which contains SDKs](https://github.com/xybp888/iOS-SDKs/blob/master/iPhoneOS15.2.sdk/System/Library/Frameworks/Metal.framework/Headers/MTLDevice.h) which shows that the iOS 15 SDK from Xcode 13 does include that header. I don't think there's any difference between the iOS and tvOS SDKs appart from the stuff that is marked as unavailable through `API_AVAILABLE`, right? But yes, that would be a good suggestion.

Reply:
Hmmm. The `MTLDevice.h` header only seems to assign availability to the `MTLGPUFamily`, and not to the individual `MTLGPUFamilyAppleN` values. 

That list currently ends with `MTLGPUFamilyApple8`. Clearly more will be added to in the future, and Xcode 14 (for instance) won't understand them when building. My concern is that `MTLGPUFamilyApple8` might not be in the `MTLGPUFamily` list in the Xcode 13 SDK.

Reply:
I have for one reported this issue to Apple through Feedback. I just downloaded Xcode 13 to look into it's SDKs and was able to find that it does in fact not have Apple8 in its macOS SDK.

Reply:
Another idea, actually, would be to have our own copy of the MTLGPUFamily enum like `MVKGPUFamily`. The enumerations are just ordinary constants which we can just reuse and that would bypass this issue entirely. @billhollings, what do you think of that?

Reply:
> @billhollings, what do you think of that?

You mean then cast our own enum values to `MTLGPUFamily` when accessing in Metal? 

I can't say I'm a fan of this approach, because the use of the `MVK_XCODE_NN` family of macros covers more than just enum values, but also method calls, object declarations, etc., that evolve with the OS.

TBH, I'm not sure I understand the complexity of these discussions. As I understand it, the original problem was that we aren't exposing formats that are available on macOS when an X86 build is running on on Apple Silicon, using Rosetta 2.

From my (admittedly academic) review, most of that can probably be handled by removing most of the invalidations at the top of `MTLPixelFormats.mm` under:

https://github.com/KhronosGroup/MoltenVK/blob/4421883eeb35d93dae8285f3c45acc66ad6c9cb6/MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.mm#L28-L30

and then removing the `MVK_MACOS_APPLE_SILICON` test here, to test the actual runtime hardware:

https://github.com/KhronosGroup/MoltenVK/blob/4421883eeb35d93dae8285f3c45acc66ad6c9cb6/MoltenVK/MoltenVK/GPUObjects/MVKPixelFormats.mm#L1492 

and possibly adding some additional `MVK_XCODE_NN` tests in that section to avoiding referencing the higher versions of `MTLGPUFamilyAppleN` on Xcode 11 & 12. 

I recognize that there are some "devils in the details" that we might encounter in an actual implementation of this, but is there much more beyond what I described?

Reply:
No and thats exactly what this PR does and I already applied the suggested changes of yours. I was just brainstorming an idea about having a custom enum because it is not entirely clear with which Xcode versions certain enumerations were added, unlike with the rest of the cases where the macros are used. But yes, that'd be something for sometime else.

Reply:
Ah. Thanks for clarifying.

My bad. I was interpreting this as still being a WIP. And yes, you're correct, the current state of this PR matches what I was suggesting above. My apologies for not picking up on that. LGTM now.